builtin/credential/aws/path_config_identity.go

```diff
@@ -5,10 +5,51 @@ import (
 	"fmt"
 
 	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/helper/authmetadata"
 	"github.com/hashicorp/vault/sdk/helper/strutil"
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
+var (
+	// iamAuthMetadataFields is a list of the default auth metadata
+	// added to tokens during login. The default alias type used
+	// by this back-end is the role ID. Subsequently, the default
+	// fields included are expected to have a low rate of change
+	// when the role ID is in use.
+	iamAuthMetadataFields = &authmetadata.Fields{
+		FieldName: "iam_metadata",
+		Default: []string{
+			"account_id",
+			"auth_type",
+		},
+		AvailableToAdd: []string{
+			"canonical_arn",
+			"client_arn",
+			"client_user_id",
+			"inferred_aws_region",
+			"inferred_entity_id",
+			"inferred_entity_type",
+		},
+	}
+
+	// ec2AuthMetadataFields is a list of the default auth metadata
+	// added to tokens during login. The default alias type used
+	// by this back-end is the role ID. Subsequently, the default
+	// fields included are expected to have a low rate of change
+	// when the role ID is in use.
+	ec2AuthMetadataFields = &authmetadata.Fields{
```

Comment:
@tyrannosaurus-becks While I know `auth_type` wasn't included previously for the ec2 logins, it seems like it should be. I can't think of a reason why we'd include this info only for IAM. Based on looking at the [original use](https://github.com/hashicorp/vault/blob/78ed5f31c61a7a108431b84c2348c18055f2eaba/builtin/credential/aws/path_login.go#L951), I suspect it was probably was intended to be added, wasn't, and because of the upgrade login its omission was never noticed.

Reply:
I agree. I will add that.

Reply:
https://github.com/hashicorp/vault/pull/8783/commits/2a6fe2ede8cbc0ed9a930dd076e97c935c3396ec